src/components/LocaleSwitcher.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div>
+    <select v-model="$i18n.locale" @change="setLocale">
+      <option value="en">English</option>
+      <option value="hi">हिंदी</option>
+    </select>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "LocaleSwitcher",
+  methods: {
+    setLocale() {
```

Comment:
It sees like we're assuming locale will be switched ONLY on `userConfig`. Just to be a little future-proof, I would suggest having two methods:

```
setLocale() {
// Simply set locale in the app
} 

setLocaleBasedOnUserConfig() {
// Fetch user config
setLocale()
}
```

Make sure all locale changing related logic is in this file only

Reply:
This method is setting the user config, or is it setting the current locale that app will show (which is what the name `LocaleSwitcher` seems to indicate. Little confused!

Reply:
Oh, let me clarify this better. So, firstly, we will be fetching and storing the user config locally before any of the language switch - either once they log in (PhoneSignIn.vue) or for logged in users this is fetched in App.vue. Then, changing the `select` box will automatically change the current locale of that app (without any other function). The change in the select box will then invoke this function which will update the user config locally and remotely. So, I don't see a need for 2 different functions unless I have not understood something that you are saying. Does this make sense?

Reply:
Still -- function is doing two things. Maybe rename to `setLocaleAndUpdateUserConfig`. 

Especially since you're updating user config, which sounds like a general operation, not only for locale. Perhaps should be in `UserProperties` component no?